fairseq/logging/progress_bar.py

```diff
@@ -310,6 +321,87 @@ def print(self, stats, tag=None, step=None):
             logger.info("{} | {}".format(self.prefix, postfix))
 
 
+try:
+    import functools
+
+    from aim import Repo as AimRepo
+
+    @functools.lru_cache()
+    def get_aim_run(repo, run_hash):
+        from aim import Run
+
+        return Run(run_hash=run_hash, repo=repo)
+
+except ImportError:
+    get_aim_run = None
+    AimRepo = None
+
+
+class AimProgressBarWrapper(BaseProgressBar):
+    """Log to Aim."""
+
+    def __init__(self, wrapped_bar, aim_repo, aim_run_hash, aim_param_checkpoint_dir):
+        self.wrapped_bar = wrapped_bar
+
+        if get_aim_run is None:
+            self.run = None
+            logger.warning("Aim not found, please install with: pip install aim")
+        else:
+            logger.info(f"Storing logs at Aim repo: {aim_repo}")
+
+            if not aim_run_hash:
+                # Find run based on save_dir parameter
+                query = f"run.checkpoint.save_dir == '{aim_param_checkpoint_dir}'"
+                try:
+                    runs_generator = AimRepo(aim_repo).query_runs(query)
+                    run = next(runs_generator.iter_runs())
+                    aim_run_hash = run.run.hash
+                except:
```

Comment:
Do we know what possible exceptions might be thrown here? Don't want to accidently catch events like memory errors, interrupts, system exit etc...

Reply:
sure, will resolve this ASAP

Reply:
@ArmenAg the range of exceptions possibly raised here might be too broad. The reason is that `Repo` class is a core of Aim SDK and covers creation/opening of repo, queries and some data collection as well. Also it depends on 3rd party libraries (`rocksdb` and `sqlite`) which can raise their own exceptions as well...
To make sure we don't miss anything, I would suggest to catch and re-raise exceptions from the following list:
`KeyboardInterrupt` `MemoryError` `SystemError` `SystemExit` `OSError`
For the rest, silently catch the exception and continue.

Reply:
I think if you do `except Exception:` this will catch all exceptions except SystemExit, KeyboardInterrupt and some other core ones (refer to this [link](https://stackoverflow.com/questions/7160983/catching-all-exceptions-in-python)). I think I'd be okay with this.

Reply:
Yes, seems `Exception` is enough generic to cover Aim and used 3rd parties. Thanks!

Reply:
Done ✅